src/utils/permissions.ts

```diff
@@ -0,0 +1,40 @@
+import { DASHBOARD_VIEWER } from './const';
+import { SupersetRole } from './role';
+import { getPermissionsByRoleID, postRequest } from './superset';
+
+export const getUserPermissions = async (
+  availableSupersetRoles: SupersetRole[],
+  headers: any,
+) => {
+  let dasboardViewerRole = availableSupersetRoles.find(
+    (ssrole: { id: number; name: string }) => ssrole.name === DASHBOARD_VIEWER,
+  );
+
+  if (!dasboardViewerRole) {
+    // Number got directly from Superset, could change
+    dasboardViewerRole = { id: 7, name: DASHBOARD_VIEWER };
```

Comment:
Let's handle the scenario of when the user does not exist and return that as an output or error to the user. This can have unintended consequences if ones don't know that this was hardcoded here.